typed/toset_test.go

```diff
@@ -244,6 +249,16 @@ var fieldsetCases = []fieldsetTestCase{{
 			_P("list", _KBF("key", "b", "id", 1), "key"),
 			_P("list", _KBF("key", "b", "id", 1), "id"),
 		)},
+		{`{"list":[{"key":"a","id":1,"nv":2},{"key":"a","id":2,"nv":3},{"key":"b","id":1},{"key":"a","id":2,"bv":true}]}`, _NS(
+			_P("list", _KBF("key", "a", "id", 1)),
+			_P("list", _KBF("key", "a", "id", 1), "key"),
+			_P("list", _KBF("key", "a", "id", 1), "id"),
+			_P("list", _KBF("key", "a", "id", 1), "nv"),
+			_P("list", _KBF("key", "a", "id", 2)),
```

Comment:
cc @jpbetz 
This code is mostly irrelevant because we only use `ToFieldSet` on objects that are applied, and we should never allow apply to have duplicates (should we change that code?). I still wanted to illustrate that when someone will update with duplicates items, we will just give them ownership of the associative key and that's it, as a way to indicate that they own that key multiple times. WDYT?

Reply:
One drawback of this, but I don't know if we can solve it anyway, is that if someone applies one item, and then it gets updated into two items, that item won't be pruned when it's removed from the applied configuration, meaning it can no longer be removed with server-side apply. I think that's reasonable.

Reply:
> This code is mostly irrelevant because we only use ToFieldSet on objects that are applied, and we should never allow apply to have duplicates (now, that being said, I'm curious if this whole change even makes sense?)

If apply can never have duplicates why are we making SMD tolerate duplicates?

Reply:
> If apply can never have duplicates why are we making SMD ToFieldSet tolerate duplicates?

Context, we're trying to solve #234

For built-ins, the `listType=map` is used a little loosely, and we don't have a great way to address that, except for accepting some duplicates. We still can't apply duplicates, but right now you can't apply an objects that has these duplicates.

> How do Updates get the list of fields to own if not via ToFieldSet?

By using "Compare", they own the fields that they changed, since we don't know exactly what they intended to own.

Reply:
> We still can't apply duplicates, but right now you can't apply an objects that has these duplicates.

Is ToFieldSet invoked on the final merged object, or the partial patch? If it is invoked on the partial patch aren't there no duplicates and this PR has no effect?

Reply:
Correct, that's more or less what I said in the PR body:
> It's not super relevant because no one should ever be able to apply duplicates

But:
1. This is for consistency, if we agree that this is how ownership of things that are duplicates, we should probably apply it everywhere, even if it's not relevant here.
2. This is the first time we use that new pattern in the code so I want to make sure we're fine with it.